src/PhotoPage.vala

```diff
@@ -535,12 +542,13 @@ public abstract class EditingHostPage : SinglePhotoPage {
             zoom_group.pack_start (zoom_original_box, false, false, 5);
 
             Gtk.Image zoom_out = new Gtk.Image.from_icon_name (Resources.ICON_ZOOM_OUT, Gtk.IconSize.MENU);
-            Gtk.EventBox zoom_out_box = new Gtk.EventBox ();
+            zoom_out_box = new Gtk.EventBox ();
             zoom_out_box.set_above_child (true);
             zoom_out_box.set_visible_window (false);
             zoom_out_box.add (zoom_out);
 
-            zoom_out_box.button_press_event.connect (on_zoom_out_pressed);
+            zoom_out_box.button_press_event.connect ((event) => on_zoom_button_pressed (event, zoom_out_box));
+            zoom_out_box.button_release_event.connect ((event) => on_zoom_button_released (event, zoom_out_box));
```

Comment:
You do not need to use a closure:
```
zoom_out_box.button_press_event.connect (zoom_button_pressed);
zoom_in_box.button_press_event.connect (zoom_button_pressed);
```
Will pass the event source to the handler. You will need to change the handler signature to:
```
private bool on_zoom_button_pressed (Gtk.Widget sender, Gdk.EventButton event) {}
```
Then cast sender to Eventbox.